WordPress/gutenberg
# Components: Shift focus into popover when opened

Partially addresses #2306

This pull request seeks to move focus to the first focusable element within a Popover when it becomes opened.

__Implementation notes:__

Included are some focus utilities for finding the next available focusable element. A few points:

- Focusable? Tabbable? Need some naming clarification
- Yes, we should probably move this to [`@wordpress/a11y`](https://www.npmjs.com/package/@wordpress/a11y) (easier to iterate here)
- Reinventing the wheel? Probably, but avoids bloat ([ally](https://github.com/medialize/ally.js) is [116kb minified, before gzip](https://wzrd.in/standalone/ally))

__Testing instructions:__

Verify that focus is moved to first focusable element when opening Inserter:

1. Navigate to Gutenberg > New Post
2. Click the Inserter
3. Note focus is shifted to the search field (and that search field behavior is otherwise unaffected, including focus constraint looping)

Verify that focus is moved to first focusable element when opening Post Visibility:

1. Navigate to Gutenberg > New Post
2. In the sidebar, click Public
3. Note focus is shifted to the first input option
